reflex/event.py

```diff
@@ -704,24 +704,41 @@ def fn():
 
 def redirect(
     path: str | Var[str],
```

Comment:
```suggestion

@overload
def redirect(
    path: str | Var[str],
    is_external: Optional[bool] = None,
    replace: bool = False,
) -> EventSpec: ...

@overload
@typing_extensions.deprecated("`external` is deprecatedm use `is_external` instead") 
def redirect(
    path: str | Var[str],
    is_external: Optional[bool] = None,
    replace: bool = False,
    external: Optional[bool] = None,
) -> EventSpec: ...

def redirect(
    path: str | Var[str],
```
nitpick beyond comprehension: this adds deprecation to the static checker